scipy/signal/filter_design.py

```diff
@@ -1057,44 +1070,84 @@ def _align_nums(nums):
         return nums
 
     except ValueError:
-        nums = list(nums)
-        maxwidth = len(max(nums, key=lambda num: atleast_1d(num).size))
+        nums = [np.atleast_1d(num) for num in nums]
+        max_width = max(num.size for num in nums)
+
+        # pre-allocate
+        aligned_nums = np.zeros((len(nums), max_width))
 
+        # Create numerators with padded zeros
         for index, num in enumerate(nums):
-            num = atleast_1d(num).tolist()
-            nums[index] = [0] * (maxwidth - len(num)) + num
+            aligned_nums[index, -num.size:] = num
 
-        return atleast_1d(nums)
+        return aligned_nums
 
 
-def normalize(b, a):
-    """Normalize polynomial representation of a transfer function.
+def normalize(num, den):
+    """Normalize numerator/denominator of a continuous-time transfer function.
 
     If values of `b` are too close to 0, they are removed. In that case, a
     BadCoefficients warning is emitted.
 
+    Parameters
+    ----------
+    num: array_like
+        Numerator of the transfer function. Can be a 2d array to normalize
+        multiple transfer functions.
+    den: array_like
+        Denominator of the transfer function. At most 1d.
+
+    Returns
+    -------
+    num: array
+        The numerator of the normalized transfer function. At least a 1d
+        array. A 2d-array if the input `num` is a 2d array.
+    den: 1d-array
+        The denominator of the normalized transfer function.
+
+    Notes
+    -----
+    Coefficients for both the numerator and denominator should be specified in
+    descending exponent order (e.g., ``s^2 + 3s + 5`` would be represented as
+    ``[1, 3, 5]``).
     """
-    b = _align_nums(b)
-    b, a = map(atleast_1d, (b, a))
-    if len(a.shape) != 1:
+    den = np.atleast_1d(den)
+    num = np.atleast_2d(_align_nums(num))
+
+    if den.ndim != 1:
         raise ValueError("Denominator polynomial must be rank-1 array.")
-    if len(b.shape) > 2:
+    if num.ndim > 2:
         raise ValueError("Numerator polynomial must be rank-1 or"
                          " rank-2 array.")
-    if len(b.shape) == 1:
-        b = asarray([b], b.dtype.char)
-    while a[0] == 0.0 and len(a) > 1:
-        a = a[1:]
-    outb = b * (1.0) / a[0]
-    outa = a * (1.0) / a[0]
-    if allclose(0, outb[:, 0], atol=1e-14):
+
+    # Trim leading zeros in denominator
+    den = np.trim_zeros(den, 'f')
+
+    # Normalize transfer function
+    num, den = num / den[0], den / den[0]
```

Comment:
With this one I disagree. Actually this notation avoids thinking about execution order all together. The reason for this is the same reason that one can do variable swapping without temporary variables in Python:
`a = 5`
`b = 6`
`a, b = b, a`
So even `den, num = den / den[0], num / num[0]` would work above.

Or is this a scipy-specific style requirement?